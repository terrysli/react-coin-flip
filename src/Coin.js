const HEADS_IMG_URL = "https://upload.wikimedia.org/wikipedia/commons/0/0a/George_Washington_Presidential_%241_Coin_obverse.png";
const TAILS_IMG_URL = "https://upload.wikimedia.org/wikipedia/commons/f/fa/Presidential_dollar_coin_reverse.png";

/**
 * Coin showing heads or tails
 *
 * Props:
 * - isHeads: true (heads) / false (tails)
 *
 * Coinflipper -> Coin
 */
function Coin({ isHeads }) {

  const coinImg = isHeads === "true" ? HEADS_IMG_URL : TAILS_IMG_URL;

  return (<img className="coin-image" src={coinImg} />);
}

export default Coin;
export {HEADS_IMG_URL, TAILS_IMG_URL};