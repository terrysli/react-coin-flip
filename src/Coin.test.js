import React from "react"
import { render } from "@testing-library/react";
import Coin, {HEADS_IMG_URL, TAILS_IMG_URL} from "./Coin";

it("renders Coin without crashing", function () {
  render(<Coin isHeads="true" />);
});

it("matches snapshot", function () {
  const { container } = render(<Coin isHeads="true" />);
  expect(container).toMatchSnapshot();
});

it("renders heads image when isHeads is true", function () {
  const { container } = render(<Coin isHeads="true" />);
  expect(container.querySelector(`img[src="${HEADS_IMG_URL}"]`)).toBeInTheDocument();
});

it("renders heads image when isHeads is false", function () {
  const { container } = render(<Coin isHeads="false" />);
  expect(container.querySelector(`img[src="${TAILS_IMG_URL}"]`)).toBeInTheDocument();
});