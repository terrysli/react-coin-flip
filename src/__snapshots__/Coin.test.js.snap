// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <img
    src="https://upload.wikimedia.org/wikipedia/commons/0/0a/George_Washington_Presidential_%241_Coin_obverse.png"
  />
</div>
`;
